enterprise/pkg/a8n/run/runner.go

```
@@ -198,6 +209,11 @@ func (r *Runner) createPlanAndJobs(
 
 	jobs := make([]*a8n.CampaignJob, 0, len(rs))
 	for _, repo := range rs {
+		externalServiceType := repo.ExternalRepo().ServiceType
+		if _, ok := SupportedExternalServices[externalServiceType]; !ok {
+			continue
```

This silently ignores unsupported repos. Would it be better to make it impossible to affect repos for which we can't make PRs? Through the search scope I suppose.

---

When manually adding changesets to campaigns, we should probably validate this too.

---

> This silently ignores unsupported repos. Would it be better to make it impossible to affect repos for which we can't make PRs? Through the search scope I suppose.

Hmm, how? This is the first filter the search results run through before the user even sees them. It's not like you see a repository and it then disappears silently from the changesets.

---

> When manually adding changesets to campaigns, we should probably validate this too.

Yeah, that makes sense. We can do this here:

https://github.com/sourcegraph/sourcegraph/blob/b4fe21deeacf0b833fa2fcd38f1706442a7abeab/enterprise/pkg/a8n/resolvers/resolver.go#L358-L360

It's a batch operation, so reporting an error for a single repository is always a bit tricky, but since this API isn't a first class feature anyway, I would say that it's okay to return an error if one of the repositories external services is not supported?